UmpleParser/src/ParseUtilities_Code.ump

```diff
@@ -0,0 +1,606 @@
+/*
+
+Copyright: All contributers to the Umple Project
+
+This file is made available subject to the open source license found at:
+http://umple.org/license
+
+ */
+
+class Token
+{
+  depend java.util.regex.*;
+  public Token(String aName, String aValue, Position aPosition)
+  {
+    this(aName,aValue);
+    setPosition(aPosition);
+  }
+
+  public static boolean isValidIdentifier(String identifier, String prefix){
+    String _prefix = "^" + prefix;
+    String body = "[a-zA-Z0-9_]*$";
+    String expression = _prefix + body;
+    if(Pattern.matches(expression, identifier)){
+    	return true;
+    }
+    return false;
+  }
+  
+  public static boolean isValidIdentifier(String identifier){
```

Comment:
If this method is used a lot over the previous one, it might be worth an explicit implementation that uses a statically compiled version of the regex: `^[a-zA-Z_][a-zA-Z\d_]*$`. It'd be more lines, but the performance boost of precompiling regex can be exponential.